tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -302,6 +302,49 @@ def dir_size():
     assert dir_size() == 0
 
 
+def test_large_manifests_passed_by_file(wandb_init, monkeypatch):
+    original_writer = (
+        wandb.sdk.interface.interface.InterfaceBase._write_artifact_manifest_file
+    )
+
+    file_written = None
+
+    def new_writer(self, manifest):
+        result = original_writer(self, manifest)
+        assert os.path.exists(result) and os.path.getsize(result) > 0
+        nonlocal file_written
+        file_written = result
+        return result
+
+    monkeypatch.setattr(
+        wandb.sdk.interface.interface.InterfaceBase,
+        "_write_artifact_manifest_file",
+        new_writer,
+    )
+    monkeypatch.setattr(
+        wandb.sdk.interface.interface,
+        "MANIFEST_FILE_SIZE_THRESHOLD",
+        0,
+    )
+
+    with wandb_init() as run:
+        artifact = wandb.Artifact(name="large-manifest", type="dataset")
+        with artifact.new_file("test_file.txt") as f:
+            f.write("test content")
+        artifact.manifest.entries["test_file.txt"].extra["test_key"] = {"x": 1}
+        run.log_artifact(artifact)
+        artifact.wait()
+
+    assert file_written is not None
+    # The file should have been cleaned up and deleted by the receiving process.
+    assert not os.path.exists(file_written)
```

Comment:
suggestion(optional): if you're OK with not _directly_ testing for the temporary existence of the `manifest_file` (on line 314):
```
assert os.path.exists(result) and os.path.getsize(result) > 0
```
...then you can probably avoid patching in `new_writer()` (and dealing with `nonlocal file_written`) here.

It looks like `pytest-mock` is a dev dependency, so you can probably use [`mocker.spy`](https://pytest-mock.readthedocs.io/en/latest/usage.html#spy) to just check that `_write_artifact_manifest_file()` was at least called:
```suggestion
def test_large_manifests_passed_by_file(wandb_init, monkeypatch, mocker):

    writer_spy = mocker.spy(
        wandb.sdk.interface.interface.InterfaceBase,
        "_write_artifact_manifest_file",
    )

    monkeypatch.setattr(
        wandb.sdk.interface.interface,
        "MANIFEST_FILE_SIZE_THRESHOLD",
        0,
    )

    with wandb_init() as run:
        artifact = wandb.Artifact(name="large-manifest", type="dataset")
        with artifact.new_file("test_file.txt") as f:
            f.write("test content")
        artifact.manifest.entries["test_file.txt"].extra["test_key"] = {"x": 1}
        run.log_artifact(artifact)
        artifact.wait()

    assert writer_spy.call_count == 1
    
    file_written = writer_spy.spy_return
    assert file_written is not None
    # The file should have been cleaned up and deleted by the receiving process.
    assert not os.path.exists(file_written)
```

Reply:
Nice! I haven't used `mocker.spy` before, it looks like a strictly superior solution here